physx/src/scene.rs

```diff
@@ -934,31 +968,89 @@ impl From<PxPairFilteringMode::Enum> for PairFilteringMode {
     }
 }
 
-#[derive(Debug, Clone, Copy, BitFlags)]
+#[derive(Default, Copy, Clone, Debug)]
+/// 0 means no limit.
+pub struct SceneLimits {
+    pub max_nb_actors: u32,
+    pub max_nb_bodies: u32,
+    pub max_nb_static_shapes: u32,
+    pub max_nb_dynamic_shapes: u32,
+    pub max_nb_aggregates: u32,
+    pub max_nb_constraints: u32,
+    pub max_nb_regions: u32,
+    pub max_nb_broad_phase_overlaps: u32,
+}
+
+impl Into<PxSceneLimits> for SceneLimits {
+    fn into(self) -> PxSceneLimits {
+        PxSceneLimits {
+            maxNbActors: self.max_nb_actors,
+            maxNbBodies: self.max_nb_bodies,
+            maxNbStaticShapes: self.max_nb_static_shapes,
+            maxNbDynamicShapes: self.max_nb_dynamic_shapes,
+            maxNbAggregates: self.max_nb_aggregates,
+            maxNbConstraints: self.max_nb_constraints,
+            maxNbRegions: self.max_nb_regions,
+            maxNbBroadPhaseOverlaps: self.max_nb_broad_phase_overlaps,
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+#[repr(u32)]
+pub enum FrictionType {
+    Patch = 0,
+    OneDirectional = 1,
+    TwoDirectional = 2,
+}
+
+impl Into<PxFrictionType::Enum> for FrictionType {
+    fn into(self) -> PxFrictionType::Enum {
+        match self {
+            FrictionType::Patch => PxFrictionType::ePATCH,
+            FrictionType::OneDirectional => PxFrictionType::eONE_DIRECTIONAL,
+            FrictionType::TwoDirectional => PxFrictionType::eTWO_DIRECTIONAL,
+        }
+    }
+}
+
+impl Default for FrictionType {
+    fn default() -> Self {
+        FrictionType::Patch
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
 #[repr(u32)]
 pub enum BroadPhaseType {
-    SweepAndPrune = 1,
-    MultiBoxPruning = 2,
-    AutomaticBoxPruning = 4,
-    GPU = 8,
+    SweepAndPrune = 0,
+    MultiBoxPruning = 1,
+    AutomaticBoxPruning = 2,
+    GPU = 3,
 }
 
 impl Into<PxBroadPhaseType::Enum> for BroadPhaseType {
     fn into(self) -> PxBroadPhaseType::Enum {
         match self {
-            BroadPhaseType::SweepAndPrune => 0,
-            BroadPhaseType::MultiBoxPruning => 1,
-            BroadPhaseType::AutomaticBoxPruning => 2,
-            BroadPhaseType::GPU => 3,
+            BroadPhaseType::SweepAndPrune => PxBroadPhaseType::eSAP,
+            BroadPhaseType::MultiBoxPruning => PxBroadPhaseType::eMBP,
+            BroadPhaseType::AutomaticBoxPruning => PxBroadPhaseType::eABP,
+            BroadPhaseType::GPU => PxBroadPhaseType::eGPU,
         }
     }
 }
 
+impl Default for BroadPhaseType {
```

Comment:
Does this kind of Default for a type even make sense? I would normally think that defaults should be specified in the default constructor of the type whose member is a BroadPhaseType in this case (basically, what I just had you change)...

I'm OK with having it though but it's unintuitive to me.